ros_buildfarm/templates/devel/devel_job.xml.em

```diff
@@ -245,6 +256,106 @@ if pull_request:
     'publisher_xunit',
     pattern='catkin_workspace/test_results/**/*.xml',
 ))@
+@[if not pull_request]@
+@(SNIPPET(
+    'publisher_groovy-postbuild',
+    script='\n'.join([
+        '// COLLATE BUILD TEST RESULTS AND EXPORT BUILD HISTORY FOR WIKI',
+        'import jenkins.model.Jenkins',
+        'import hudson.FilePath',
+        '',
+        '@Grab(\'org.yaml:snakeyaml:1.17\')',
+        'import org.yaml.snakeyaml.Yaml',
+        'import org.yaml.snakeyaml.DumperOptions',
+        '',
+        'manager.listener.logger.println("# BEGIN SECTION: Collate test results for wiki.")',
+        '',
+        '// nr of builds to include in history',
+        'final num_build_hist = 5',
+        '',
+        'try {',
+        '  def data = [',
+        '    "history" : []',
+        '  ]',
+        '',
+        '  // gather info on tests of current build',
+        '  def tresult = manager.build.getAction(hudson.tasks.junit.TestResultAction.class)?.result',
+        '  if (tresult)',
+        '  {',
+        '    data.latest_build = [',
+        '      "skipped" : tresult.skipCount,',
+        '      "failed" : tresult.failCount,',
+        '      "total" : tresult.totalCount',
+        '    ]',
+        '  }',
+        '  else',
+        '  {',
+        '    manager.listener.logger.println("No test result action for last build, skipping gathering statistics for it.")',
+        '  }',
+        '',
+        '',
+        '  // get acces to the job of the running build',
+        '  def job_name = manager.build.getEnvironment(manager.listener).get(\'JOB_NAME\')',
+        '  manager.listener.logger.println("Collating test statistics for \'${job_name}\'.")',
+        '  def job = Jenkins.instance.getItem(job_name)',
+        '  if (job == null)',
+        '  {',
+        '    manager.listener.logger.println("No such job: \'${job_name}\'.")',
+        '    return',
+        '  }',
+        '',
+        '  // store base info',
+        '  data.base_url = Jenkins.instance.getRootUrl()',
+        '  data.total_builds = job.builds.size()',
+        '  data.job_health = job.getBuildHealth().getScore()',
+        '  data.job_health_icon = job.getBuildHealth().getIconClassName()',
+        '',
+        '  // retrieve info on last N builds of this job',
+        '  job.builds.take(num_build_hist).each { b ->',
+        '    tresult = b.getAction(hudson.tasks.junit.TestResultAction.class)?.result',
+        '    if (tresult)',
+        '    {',
+        '      data.history << [',
+        '        "job_id" : b.id as Integer,',
```

Comment:
`job_id` doesn't sounds right to. I would propose either `build_id` or `build_number`.